[Shamus] iP

shamooose opened the pull request:
### **Scruffles**

Scruffles frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉
5. And it is FREE!

Features:

- [x] Managing tasks
- [x]  Managing deadlines
- [ ]  Reminders (coming soon)

Commit:
Add Gradle support

shamooose committed:
Changed name of ip, added custom greeting, added echoes and exit message

shamooose committed:
Revert "Changed name of ip, added custom greeting, added echoes and exit message"

This reverts commit bacce96e427444de06461ed5564ebc48627123e4.

shamooose committed:
Revert "Revert "Changed name of ip, added custom greeting, added echoes and exit message""

This reverts commit 6a4094175696583b3467281c3850c25c4706bbbc.

shamooose committed:
Added the ability to store whatever text entered by the user and display them back to the user when requested.

shamooose committed:
Added the ability to mark tasks as done.

shamooose committed:
Added support for tracking ToDos, Deadlines and Events.

shamooose committed:
Handles exceptions and errors now

shamooose committed:
Added support for deleting tasks from the list.

shamooose committed:
Used the input/output redirection technique to semi-automate the testing of Duke.

shamooose committed:
Saves the tasks in the hard disk automatically whenever the task list changes.

shamooose committed:
Understands dates and times.

onepersonhere commented on src/main/java/Scruffles.java:
```diff
@@ -0,0 +1,225 @@
+            Task task = new Task("");
+            boolean isDone = inputArray[1].equals("X");
+
+            switch (inputArray[0]) {
+                case "T":
+                    task = new Todo(inputArray[2], isDone);
+                    break;
+                case "D":
+                    task = new Deadline(inputArray[2], LocalDate.parse(inputArray[3]), isDone);
+                    break;
+                case "E":
+                    task = new Event(inputArray[2], LocalDate.parse(inputArray[3]), LocalTime.parse(inputArray[4]),
+                            LocalTime.parse(inputArray[5]), isDone);
+                    break;
+                default:
+                    continue;
```
Perhaps the switch statement should have the following form: No indentation for case clauses?

onepersonhere commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,23 @@
+import java.time.LocalDate;
```
Perhaps every class should be part of some package? I noticed the same issue in all of your .java files.

onepersonhere commented on src/main/java/Scruffles.java:
```diff
@@ -0,0 +1,225 @@
+                } catch (DescriptionEmptyException e) {
+                    System.out.println(e.getMessage());
+                } catch (DateTimeParseException e) {
+                    System.out.println("grrrr >:( please input event date as 'yyyy-mm-dd from hh:mm to hh:mm' format woof woof!");
```
Perhaps line length should be no longer than 120 chars?

onepersonhere commented on src/main/java/Scruffles.java:
```diff
@@ -0,0 +1,225 @@
+                        throw new DescriptionEmptyException("grrrr >:( when is your deadline?? woof woof!");
+                    } else {
+                        input = input.replaceFirst("deadline ", "");
+                        String[] inputArray = input.split(" /by ");
```
Perhaps plural form should be used on names representing a collection of objects?
```suggestion
                        String[] inputs = input.split(" /by ");
```

onepersonhere commented on src/main/java/Scruffles.java:
```diff
@@ -0,0 +1,225 @@
+                        } else if (task instanceof Deadline) {
+                            textInput = "D / " + isDone + task.taskName + " / " + ((Deadline) task).by + "\n";
+                        } else if (task instanceof Event) {
+                            textInput = "E / " + isDone + task.taskName + " / " + ((Event) task).at + " / "
+                                    + ((Event) task).startTime + " / " + ((Event) task).endTime + "\n";
```
Perhaps place more line breaks to improve readability?

onepersonhere reviewed (commented):
Overall, I found your code hard to read for the most part where the nesting was too deep. 
I noted a few minor coding standard violations too. 
Some of the classes are getting quite long. 
Consider splitting into smaller classes if that makes sense.

shamooose committed:
Made the code more OOP

shamooose committed:
Organized the classes into 1 package

shamooose committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

shamooose committed:
Added checkstyle plugin to gradle

shamooose committed:
Added JUnit tests to test the behavior of the code.

shamooose committed:
Packaged the app as a JAR file

shamooose committed:
Added Javadoc comments

shamooose committed:
Tweaked the code to comply with the coding standard

shamooose committed:
Give users a way to find a task by searching for a keyword.

shamooose committed:
Added GUI to Scruffles. Used JavaFX Technology to implement the GUI.

shamooose committed:
Reduce method length for add

The code is currently too long and makes it less readable.

Shorten the method and add more methods.

This helps to make the code more readable.

Let's add a switch to my add method and create a todo, deadline and event method

shamooose committed:
Debug the saving and loading of data

Currently, the saving and loading of data does not work as the saveToFile method saves an empty text file.

Debug the code by allowing the Parser to load the text file when initiated.

This allows the program to load the previously saved data easily and create a new TaskList when there is no data.

shamooose committed:
Add assertions

shamooose committed:
Improve code quality

shamooose committed:
Provide support for managing tasks that need to be done within a certain period

shamooose committed:
Clean up code quality and add personality

Deep nesting in some methods amkes the code less readable.

Let's refactor the code to eliminate deep nesting.

The code is now more readable.

shamooose committed:
Add Ui.png to docs

shamooose committed:
Add a user guide

shamooose committed:
Update test case for Event

shamooose committed:
Use checkStyle to detect coding style violations.

shamooose committed:
Implement changes from CheckStyle

shamooose committed:
Remove unused code

shamooose committed:
Merge pull request #2 from shamooose/test

Remove unused code
